packages/react/src/lib/accessibility/FocusZone/AutoFocusZone.tsx

```diff
@@ -0,0 +1,245 @@
+        onKeyDown={this._onKeyboardHandler}
+        onFocusCapture={this._onFocusCapture}
+      >
+        {this.props.children}
+      </ElementType>
+    )
+  }
+
+  public componentWillUnmount(): void {
+    AutoFocusZone._focusStack = AutoFocusZone._focusStack.filter((value: AutoFocusZone) => {
+      return this !== value
+    })
+
+    this._unsubscribeFromEvents()
+  }
+
+  private _findElementAndFocusAsync = () => {
+    if (!this._root.current) return
+    const { focusPreviouslyFocusedInnerElement, firstFocusableSelector } = this.props
+
+    if (
+      focusPreviouslyFocusedInnerElement &&
+      this._previouslyFocusedElementInTrapZone &&
+      this._root.current.contains(this._previouslyFocusedElementInTrapZone)
+    ) {
+      // focus on the last item that had focus in the zone before we left the zone
+      focusAsync(this._previouslyFocusedElementInTrapZone)
+      return
+    }
+
+    const focusSelector =
+      firstFocusableSelector &&
+      (typeof firstFocusableSelector === 'string'
+        ? firstFocusableSelector
+        : firstFocusableSelector())
+
+    const firstFocusableChild = focusSelector
+      ? (this._root.current.querySelector(`.${focusSelector}`) as HTMLElement)
+      : getNextElement(
+          this._root.current,
+          this._root.current.firstChild as HTMLElement,
+          true,
+          false,
+          false,
+          true,
+        )
+
+    firstFocusableChild && focusAsync(firstFocusableChild)
+  }
+
+  private _onFocusCapture = (ev: React.FocusEvent<HTMLDivElement>) => {
+    this.props.onFocusCapture && this.props.onFocusCapture(ev)
+    if (ev.target !== ev.currentTarget) {
+      // every time focus changes within the trap zone, remember the focused element so that
+      // it can be restored if focus leaves the pane and returns via keystroke (i.e. via a call to this.focus(true))
+      this._previouslyFocusedElementInTrapZone = ev.target as HTMLElement
+    }
+  }
+
+  private _onKeyboardHandler = (ev: React.KeyboardEvent<HTMLDivElement>): void => {
+    this.props.onKeyDown && this.props.onKeyDown(ev)
+
+    if (
+      ev.isDefaultPrevented() ||
+      keyboardKey.getCode(ev) !== keyboardKey.Tab ||
+      !this._root.current
+    ) {
+      return
+    }
+
+    const _firstTabbableChild = getFirstTabbable(
+      this._root.current,
+      this._root.current.firstChild as HTMLElement,
+      true,
+    )
+    const _lastTabbableChild = getLastTabbable(
+      this._root.current,
+      this._root.current.lastChild as HTMLElement,
+      true,
+    )
+
+    if (ev.shiftKey && _firstTabbableChild === ev.target) {
+      focusAsync(_lastTabbableChild)
+      ev.preventDefault()
+      ev.stopPropagation()
+    } else if (!ev.shiftKey && _lastTabbableChild === ev.target) {
+      focusAsync(_firstTabbableChild)
+      ev.preventDefault()
+      ev.stopPropagation()
+    }
+  }
+
+  private _forceFocusInTrap = (ev: Event, triggeredElement: HTMLElement) => {
```

Comment:
We might not need this,  as `AutoFocusZone` should just focus inner element on mount, it should be still possible to TAB out from it